client/state/jetpack-connect/actions.js

```diff
@@ -380,10 +379,7 @@ export function authorize( queryObject ) {
 					} )
 				);
 				debug( 'Site updated', data );
-				dispatch( {
-					type: SITE_RECEIVE,
-					site: data,
-				} );
+				receiveSite( data );
```

Comment:
The action creator itself won't do the job; it needs to be `dispatch()`-ed, so:

```
receiveSite( data );
```

needs to become:

```
dispatch( receiveSite( data ) );
```

Reply:
🤦‍♂️ Thanks!